### PR TITLE
Check that the extracted text has a described by link

### PR DESCRIPTION
**Pull request by kalaspuffar**
Trying to solve https://github.com/nlbdev/nordic-epub3-dtbook-migrator/issues/519

### CODE REVIEW

File: src/main/resources/xml/schema/2020-1/nordic2020-1.sch [hunk 1]
```diff
@@ -630,6 +630,18 @@
         </rule>
     </pattern>
 
+    <pattern id="epub_nordic_253_d">
+        <title>Rule 253d</title>
+        <p>Extracted text needs to have a reference of the image the text have been extracted from.</p>
+        <rule context="html:figure[tokenize(@class, '\s+') = 'image']">
+            <let name="figdesc-id" value=".//html:aside[tokenize(@class, '\s+') = 'fig-desc']/@id" />
+            <let name="aria-id" value=".//html:img/@aria-describedby" />
+            <assert test="count(.//html:aside[tokenize(@class, '\s+') = 'fig-desc']) = 0 or $figdesc-id = $aria-id">
```
Comment (josteinaj) [inline src/main/resources/xml/schema/2020-1/nordic2020-1.sch:639]: If both attributes are missing or empty, I think this assetion will succeed? This assertion covers the second check described by Martin in #519, but the first check also needs to be implemented. That is, img elements in this context needs to have a non-empty aria-describedby attribute.

**Review by josteinaj — CHANGES_REQUESTED** [review]

File: src/main/resources/xml/schema/2020-1/nordic2020-1.sch (hunk 1 above)
Comment (kalaspuffar) [reply src/main/resources/xml/schema/2020-1/nordic2020-1.sch:639]: Hi @josteinaj 

I'm not sure if I understand the finer details here, but if I look at https://format.mtm.se/nordic_epub/2020-1#text-extraction-from-images it seems like the aria-describedby is only required in the context that we have, a fig-desc (extracted text) element. And this rule will verify both that it's present and that it's equal to the description.

Is there a nuance I've missed and could you give me an example?

Best regards
Daniel

**Comment by kalaspuffar** [comment]
Hi @josteinaj 

I've added the test code I ran in the old repository. I now have this repository setup to run the java test directly in this repository. We might want to remove the generated XSLT files, though, to ease the review process. I'll look into that.

Best regards
Daniel

**Comment by kalaspuffar** [comment]
I'm sorry that this PR is a bit messy but future ones will be more clean.

**Comment by josteinaj** [comment]
After thinking about it, this looks good. Are we ready to merge?

**Review by josteinaj — APPROVED** [review]

**Comment by kalaspuffar** [comment]
Hi @josteinaj 

Yes, this PR is ready to merge. I will create another convenience PR as well that you might want to merge later.

Best regards
Daniel

**Comment by kalaspuffar** [comment]
Hi @erikmtm

Do you want to review this as well?

Best regards
Daniel

**Comment by josteinaj** [comment]
I'm merging this.

@erikmtm: if you have comments, we can always change it later :+1:

**Pull request merged**